modules/repository/src/main/java/org/mule/runtime/module/repository/api/RepositoryService.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.repository.api;
+
+import java.io.File;
+
+/**
+ * Service that provides access to bundle. A bundle may be a jar, zip or any type supported type by the platform.
+ *
+ * The repository may access just the local file system or an external service to download the bundles. The service will
+ * use a local file system repository in order to avoid download bundles that have already being downloaded or pre-installed
+ * in the server.
+ *
+ *
+ *
+ * @since 4.0
+ */
+public interface RepositoryService
+{
+
+    /**
+     * System property key to specify a custom repository folder. By default the container will use $MULE_HOME/lib/repository
+     */
+    String MULE_REPOSITORY_FOLDER_PROPERTY = "mule.repository.folder";
```

Comment:
As those properties are meant to be used by the end user of this feature it makes sense to be part of the API.